src/sys/mman.rs

```diff
@@ -187,6 +190,31 @@ libc_bitflags! {
     }
 }
 
+impl MapFlags {
+    /// Create `MAP_HUGETLB` with provided size of huge page.
+    ///
+    /// Under the hood it computes `MAP_HUGETLB | (huge_page_size_log2 << libc::MAP_HUGE_SHIFT`).
+    /// `huge_page_size_log2` denotes logarithm of huge page size to use and should be
+    /// between 16 and 63 (inclusively).
+    ///
+    /// ```
+    /// # use nix::sys::mman::MapFlags;
+    /// let f = MapFlags::map_hugetlb_with_size_log2(30).unwrap();
+    /// assert_eq!(f, MapFlags::MAP_HUGETLB | MapFlags::MAP_HUGE_1GB);
+    /// ```
+    #[cfg(target_os = "linux")]
```

Comment:
This should probably include android and fuchsia, too.  At least, libc defines MAP_HUGETLB for those platforms.  But it doesn't define MAP_HUGE_SHIFT.  Is that an oversight in libc?

Reply:
>Is that an oversight in libc?

Maybe. Either way, we would need first introduce the constant to `libc` and only then add expand list of targets, so I think it's better to do it in a later PR.

Reply:
I filed a PR adding these symbols for Android or Linux: https://github.com/rust-lang/libc/pull/3444